waiverdb/api_v1.py

```diff
@@ -730,10 +705,15 @@ def get(self):
         """
         args = GetPermissions.parse_obj(request.args)
         testcase = args.testcase
+        permissions_to_ret = permissions()
         if testcase:
-            return list(match_testcase_permissions(testcase, permissions()))
-
-        return permissions()
+            permissions_to_ret = list(match_testcase_permissions(testcase, permissions_to_ret))
+        if not args.html:
```

Comment:
This should be probably handled checking "Accept" request HTTP header instead. Web browser should normally use something like `Accept: text/html,...` in the requests.

```python
if request.accept_mimetypes.accept_html:
    return Response(
        render_template('permissions.html', permissions=permissions_to_ret),
        mimetype='text/html'
    )

return permissions_to_ret
```

One inconvenience can be that both `curl` and web browsers can also include `*/*` in  "Accept" header by defaul, so I think with `curl` you would need to explicitly request JSON or perhaps use the following check instead:

```python
if "text/html" in request.headers.get("Accept", ""):
    ...
```

Reply:
I would better stay backward-compatible, i.e. if someone uses an API call to permissions and expect to receive a JSON, they would still receive it unless, explicitly specified the `html=1` as an argument

Reply:
OK